tests/build-tests/components.test.ts

```diff
@@ -0,0 +1,106 @@
+import { execa } from 'execa';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { afterEach, beforeEach, describe, it } from 'vitest';
+
+import { AddonFixtureHelper } from '../fixture-helper.js';
+import { createAddon, createTmp, install } from '../utils.js';
+
+let packageManager = 'pnpm';
+
+describe('components (build)', () => {
+  describe('co-located JS', () => {
+    let cwd = '';
+    let tmpDir = '';
+    let addonOnlyJS: AddonFixtureHelper;
+
+    beforeEach(async () => {
+      tmpDir = await createTmp();
+
+      console.debug(`Debug test repo at ${tmpDir}`);
+
+      let { name } = await createAddon({
+        args: [`--pnpm=true`, '--addon-only'],
```

Comment:
Why are we doing this for `addon-only` here? When setting up new tests, shouldn't we just start with the default scenario first?

Reply:
related to [here](https://github.com/embroider-build/addon-blueprint/pull/154#discussion_r1264688802) i was focusing on compiled output, rather than e2e "workingness", if that makes sense -- I'm going to shift to the default scenario tho, and make sure that each type of component can be rendered in the test-app